metaflow/plugins/kfp/kfp.py

```diff
@@ -263,30 +264,70 @@ def _create_workflow_yaml(
 
         elif output_format == "argo-workflow-template":
             workflow["kind"] = "WorkflowTemplate"
+            workflow["spec"]["serviceAccountName"] = (
+                KUBERNETES_SERVICE_ACCOUNT or "default-editor"
+            )
 
             # Use static name to make referencing easier.
             # Note the name has to follow k8s format.
             # self.name is typically CamelCase as it's python class name.
             # generateName contains a sanitized version of self.name from kfp.compiler
-            workflow["metadata"]["name"] = (
+            workflow_name = (
                 name if name else workflow["metadata"].pop("generateName").rstrip("-")
             )
+            workflow["metadata"]["name"] = workflow_name
+
+            workflow["spec"]["synchronization"] = {
+                "semaphore": {
+                    "configMapKeyRef": {"name": workflow_name, "key": "max_concurrency"}
+                }
+            }
 
-            workflow["spec"]["serviceAccountName"] = (
-                KUBERNETES_SERVICE_ACCOUNT or "default-editor"
-            )
         else:
             raise NotImplementedError(f"Unsupported output format {output_format}.")
 
         return workflow
 
+    @staticmethod
+    def _create_config_map(workflow_name: str, max_concurrency: int):
+        config_map = {
+            "apiVersion": "v1",
+            "kind": "ConfigMap",
+            "metadata": {"name": workflow_name},
+            "data": {"max_concurrency": str(max_concurrency)},
+        }
+        return config_map
+
+    @staticmethod
+    def _create_cron_workflow(
+        name: str,
+        schedule: Optional[str] = None,
+        concurrency: Optional[str] = None,
+        recurring_run_enable: Optional[bool] = False,
+    ) -> Dict[str, Any]:
+        body = {
+            "apiVersion": "argoproj.io/v1alpha1",
+            "kind": "CronWorkflow",
+            "metadata": {"name": name},
+            "spec": {
+                "suspend": not recurring_run_enable,
+                "schedule": (
+                    schedule if schedule else "* * 0 * *"
```

Comment:
Do we want to keep using 6 digit cron?

Reply:
it's a five place cron (which excludes seconds):
see
* https://argoproj.github.io/argo-workflows/cron-workflows/
* https://pkg.go.dev/github.com/robfig/cron#hdr-CRON_Expression_Format

go cron allows 6 space (seconds).

This brings up a good call, to not allow an interval less than 15 minutes! created https://zbrt.atl.zillow.net/browse/AIP-7375 _argo->kfp: Limit recurring run to > 15 mins_ to be done in the webhook as a policy